static/main.ts

```diff
@@ -0,0 +1,126 @@
+};
+
+const alchemyKey = ALCHEMY_KEY || "";
+
+const providersUrl: { [key: string]: string } = {
+  31337: `http://localhost:8545`,
+  100: alchemyKey ? `https://gnosis-mainnet.g.alchemy.com/v2/${alchemyKey}` : "https://rpc.gnosischain.com",
+  1: alchemyKey ? `https://eth-mainnet.alchemyapi.io/v2/${alchemyKey}` : "https://eth.llamarpc.com",
+  137: alchemyKey ? `https://polygon-mainnet.g.alchemy.com/v2/${alchemyKey}` : "https://polygon.llamarpc.com",
+  10: alchemyKey ? `https://opt-mainnet.g.alchemy.com/v2/${alchemyKey}` : "https://optimism.llamarpc.com",
+  42161: alchemyKey ? `https://arb-mainnet.g.alchemy.com/v2/${alchemyKey}` : "https://arbitrum.llamarpc.com",
+  8453: alchemyKey ? `https://base-mainnet.g.alchemy.com/v2/${alchemyKey}` : "https://base.llamarpc.com",
+  56: alchemyKey ? `https://bnb-mainnet.g.alchemy.com/v2/${alchemyKey}` : "https://binance.llamarpc.com",
+  81457: alchemyKey ? `https://blast-mainnet.g.alchemy.com/v2/${alchemyKey}` : "https://blast.drpc.org",
+  324: alchemyKey ? `https://zksync-mainnet.g.alchemy.com/v2/${alchemyKey}` : "https://mainnet.era.zksync.io",
+  43114: alchemyKey ? `https://avax-mainnet.g.alchemy.com/v2/${alchemyKey}` : "https://rpc.ankr.com/avalanche",
+  480: alchemyKey ? `https://worldchain-mainnet.g.alchemy.com/v2/${alchemyKey}` : "https://rpc.worldchain.network",
+};
+
+let networks: [AppKitNetwork, ...AppKitNetwork[]];
+if (window.location.hostname === "localhost" || window.location.hostname === "0.0.0.0") {
+  console.log("enabling anvil");
+  networks = [anvil, gnosis, mainnet, polygon, optimism, arbitrum, base, bsc, blast, zksync, avalanche, worldchain];
+} else {
+  networks = [gnosis, mainnet, polygon, optimism, arbitrum, base, bsc, blast, zksync, avalanche, worldchain];
+}
+
+export const appState = createAppKit({
+  adapters: [new Ethers5Adapter()],
+  networks,
+  defaultNetwork: gnosis,
+  metadata,
+  projectId,
+  features: {
+    analytics: true,
+  },
+});
+
+// get network
+function getNetwork(): string {
+  const network = appState.getCaipNetwork();
+  return network ? network.id.toString() : "";
+}
+
+// create provider & signer for gnosis
+export let provider = new ethers.providers.JsonRpcProvider(providersUrl[getNetwork()]);
+export let userSigner: ethers.Signer;
+
+async function waitForConnection() {
+  return new Promise<void>((resolve) => {
+    const interval = setInterval(() => {
+      if (appState.getIsConnectedState()) {
+        userSigner = provider.getSigner(appState.getAddress());
+        console.log(`user connected: ${appState.getAddress()}`);
+        clearInterval(interval);
+        resolve();
+      } else {
+        console.log("waiting for user to connect...");
+      }
+    }, 1000);
+  });
+}
+
+function handleNetworkSwitch() {
+  appState.subscribeCaipNetworkChange(async (newNetwork: { id: string; name: string }) => {
+    switchProvider(newNetwork?.id);
+    updateTokens(); // update known tokens on the dropdown
```

Comment:
```suggestion
    updateTokenDropdown();
```
maybe this is a better name to avoid explaining in the comment

Reply:
I agree!